pkg/observer/cache.go

```diff
@@ -0,0 +1,59 @@
+package observer
+
+import (
+	"fmt"
+
+	"github.com/cilium/tetragon/pkg/api/dataapi"
+	lru "github.com/hashicorp/golang-lru/v2"
+)
+
+type Cache struct {
```

Comment:
Everything here is in one package, so no need to export. Could you make it `cache` and `newCache()`, so that they're not used from other packages accidentally?

Reply:
Good point. I made these changes in recent commit.